roles/kubernetes/control-plane/tasks/kubeadm-setup.yml

```diff
@@ -92,6 +92,34 @@
   set_fact:
     kubeadmConfig_api_version: v1beta3
 
+- name: Kubeadm | Patch kube-scheduler manifest for scheduler-plugins
+  when:
+    - kube_major_version is version('v1.28', '<')
+    - scheduler_plugins_enabled is defined and scheduler_plugins_enabled
+    - scheduler_plugins_single_scheduler_mode is defined and scheduler_plugins_single_scheduler_mode
+  tags:
+    - scheduler_plugins
+  block:
+    - name: Enable kubeadm_patches
```

Comment:
Instead of modifying the fact, I think you should rather reuse the existing kubeadm_patches task ("Kubeadm | Create directory to store kubeadm patches") adding the condition to fire it if we need it for the scheduler.
Also, you don't need to add the patch to localhost, just push it directly to the destination.
We should avoid messing with `kubeadm_patches.source/dest_dir`, (because they are exposed to user)